pkg/extension/sumologicextension/extension.go

```diff
@@ -268,6 +281,63 @@ func (pm *sumologicExtension) register(ctx context.Context, collectorName string
 	return nil
 }
 
+func (pm *sumologicExtension) heartbeat(ctx context.Context) {
+	heartbeatInterval := defaultHeartbeatInterval
+	if pm.conf.HeartBeatInterval > 0 {
+		heartbeatInterval = pm.conf.HeartBeatInterval
+	}
+	if pm.registrationInfo.CollectorCredentialId == "" || pm.registrationInfo.CollectorId == "" {
+		pm.logger.Error("Collector not registered, cannot send heartbeat")
+		return
+	}
+	u, err := url.Parse(pm.baseUrl + heartbeatUrl)
+	if err != nil {
+		pm.logger.Error("Unable to parse heartbeat URL ", zap.String("error: ", err.Error()))
+		return
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), nil)
+	if err != nil {
+		pm.logger.Error("Unable to create HTTP request. ", zap.String("error: ", err.Error()))
+		return
+	}
+
+	pm.addCollectorCredentials(req)
+	pm.addJSONHeaders(req)
+
+	pm.logger.Info("Heartbeat heartbeat API initialized. Starting sending hearbeat requests")
+	for {
+		select {
+		case <-pm.ch:
+			pm.logger.Info("Heartbeat sender turn off")
+			return
+		default:
+			res, err := http.DefaultClient.Do(req)
+			if err != nil {
+				pm.logger.Error("Unable to send HTTP request. ", zap.String("error: ", err.Error()))
+				return
+			}
+			defer res.Body.Close()
+			if res.StatusCode != 204 {
+				var buff bytes.Buffer
+				if _, err := io.Copy(&buff, res.Body); err != nil {
+					pm.logger.Error(
+						"failed to copy collector heartbeat response body, status code: %d, err: %w",
+						zap.Int("response status code", res.StatusCode),
+						zap.String("error: ", err.Error()))
+					return
+				}
+				pm.logger.Error("Collector heartbeat request failed",
+					zap.Int("response status code", res.StatusCode),
+					zap.String("response", buff.String()),
+				)
+				return
+			}
+			pm.logger.Debug("Heartbeat sent")
+			time.Sleep(heartbeatInterval * time.Second)
```

Comment:
Assign a proper default (like suggested elsewhere: `15 * time.Second`) so that we don't need to specify `... * time.Second`

```suggestion
			time.Sleep(heartbeatInterval)
```